app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -291,13 +291,18 @@ def create_vm(args = {})
       vm
     end
 
-    def preallocate_disks(args)
-      change_allocation_volumes = args[:volumes_attributes].values.select { |x| x[:id].present? && x[:preallocate] == '1' }
-      if args[:template].present? && change_allocation_volumes.present?
-        disks = change_allocation_volumes.map do |volume|
-          { :id => volume[:id], :sparse => 'false', :format => 'raw', :storagedomain => volume[:storage_domain] }
+    def preallocate_and_clone_disks(args, template)
+      change_volumes = args[:volumes_attributes].values.select { |x| x[:id].present? }
```

Comment:
IMHO `volumes_to_change` or `existing_volumes` sounds a bit better

Reply:
`existing_volumes` is confusing here because they are new for this vm, they are just populated from the template.
`volumes_to_change` sounds a lot better.